File: Tests/CK.TypeScript.Tests/CrisLike/Model/ICommandAuthDeviceId.cs
using CK.Core;

namespace CK.CrisLike;

/// <summary>
/// Extends the basic <see cref="ICommandAuthUnsafe"/> to add the <see cref="DeviceId"/> field.
/// </summary>
[CKTypeDefiner]
public interface ICommandAuthDeviceId : ICommandAuthUnsafe
{
    /// <summary>
    /// Gets or sets the device identifier.
    /// </summary>
    [UbiquitousValue]
    string DeviceId { get; set; }
}

File: Tests/CK.TypeScript.Tests/TSInlineTests/CrisLike_commands_and_results/ck-gen/CK/CrisLike/ICommandAuthDeviceId.ts
import { ICommandAuthUnsafe } from './ICommandAuthUnsafe';

/**
 * Extends the basic ICommandAuthUnsafe to add the 
 * ICommandAuthDeviceId.deviceId field.
 **/
export interface ICommandAuthDeviceId extends ICommandAuthUnsafe {
/**
 * Gets or sets the device identifier.
 **/
deviceId: string;
readonly _brand: ICommandAuthUnsafe["_brand"] & {"29":any};
}
